utils/pattern/src/frontend/mod.rs

```diff
@@ -76,134 +65,79 @@ use writeable::{adapters::TryWriteableInfallibleAsWriteable, PartsWrite, TryWrit
-)]
-pub struct Pattern<Backend, Store: ?Sized> {
-    _backend: PhantomData<Backend>,
-    store: Store,
+#[repr(transparent)]
+pub struct Pattern<B: PatternBackend> {
+    _backend: PhantomData<B>,
+    /// The encoded storage
+    pub store: B::Store,
 }
 
-impl<Backend, Store> Pattern<Backend, Store> {
-    pub fn take_store(self) -> Store {
-        self.store
-    }
-
-    /// Creates a pattern from a serialized backing store without checking invariants.
-    /// Most users should prefer [`Pattern::try_from_store()`].
-    ///
-    /// The store is expected to come from a valid `Pattern` with this `Backend`,
-    /// such as by calling [`Pattern::take_store()`]. If the store is not valid,
-    /// unexpected behavior may occur.
-    ///
-    /// To parse a pattern string, use [`FromStr::from_str`].
-    ///
-    /// # Examples
-    ///
-    /// ```
-    /// use core::str::FromStr;
-    /// use icu_pattern::Pattern;
-    /// use icu_pattern::SinglePlaceholder;
-    /// use writeable::assert_writeable_eq;
-    ///
-    /// // Create a pattern from a valid string:
-    /// let allocated_pattern =
-    ///     Pattern::<SinglePlaceholder, String>::from_str("{0} days")
-    ///         .expect("valid pattern");
-    ///
-    /// // Transform the store and create a new Pattern. This is valid because
-    /// // we call `.take_store()` and `.from_store_unchecked()` on patterns
-    /// // with the same backend (`SinglePlaceholder`).
-    /// let store = allocated_pattern.take_store();
-    /// let borrowed_pattern: Pattern<SinglePlaceholder, &str> =
-    ///     Pattern::from_store_unchecked(&store);
-    ///
-    /// assert_writeable_eq!(borrowed_pattern.interpolate([5]), "5 days");
-    /// ```
-    pub const fn from_store_unchecked(store: Store) -> Self {
-        Self {
-            _backend: PhantomData,
-            store,
-        }
+impl<B: PatternBackend> PartialEq for Pattern<B> {
+    fn eq(&self, other: &Self) -> bool {
+        self.store == other.store
     }
 }
 
-impl<Backend, Store: ?Sized> Pattern<Backend, Store> {
-    /// Creates a `&Pattern` from a `&Store` without checking invariants.
-    pub const fn from_ref_store_unchecked(store: &Store) -> &Self {
-        // Safety: Pattern's layout is the same as `Store`'s
-        unsafe { &*(store as *const Store as *const Self) }
+impl<B: PatternBackend> core::fmt::Debug for Pattern<B> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("Pattern")
+            .field("_backend", &self._backend)
+            .field("store", &&self.store)
+            .finish()
     }
 }
 
-impl<B, Store> Pattern<B, Store>
+#[cfg(feature = "alloc")]
+impl<B: PatternBackend> ToOwned for Pattern<B>
 where
-    B: PatternBackend,
-    Store: AsRef<B::Store>,
+    Box<B::Store>: for<'a> From<&'a B::Store>,
 {
-    /// Creates a pattern from a serialized backing store.
-    ///
-    /// To parse a pattern string, use [`FromStr::from_str`].
-    ///
-    /// # Examples
```

Comment:
Nit: Bring back the example, please

Reply:
This is a moot examples because users don't have to do store-juggling anymore.